### ISSUE TITLE
Add support for Pebble smart watch

### ISSUE THREAD

**Issue by LimitlessLED** [eliotstocker/Light-Controller#32]
Home Remote has recently added iPhone support for Pebble smart watch.  Would be awesome if Light-Controller app could support pebble from an Android perspective!  there are plenty of Pebble owners with Android.  Pebble just announced the 1millionth Pebble watch sold.. that is by far more than any android gear watch.

Cheers,
Hamish.

**Comment by eliotstocker** [comment]
ill look into this once the android wear implementation is done.

**Comment by mrwhale** [comment]
I did have a look at starting an app for pebble! But got lost in documentation, then got distracted haha @LimitlessLED do you have a beta app that you could share with us that we could work off? Off so I wouldn't mind getting stuck into it

**Comment by mnwalker** [comment]
Hi all, I have been following this project for a while as I was planning to build something similar - was planning a mobile app combined with a Pi based controller/server in order to provide remote access and IFTTT support. Been too busy on other projects to get it started but seeing the progress and teamwork on this app would probably be better to join forces rather than compete.

Just wanted to chip in on this topic to mention that I have a pebble in case assistance is needed on testing or development.

**Comment by eliotstocker** [comment]
The More the merrier, always happy for some more hands :)

**Comment by LimitlessLED** [comment]
Yes I have sourcecode for the iPhone to pebble communications I can post, but it is not relevant enough for an android implementation.  Is there any similar github pebble projects with home remote control features for android?

**Comment by LimitlessLED** [comment]
I wonder if it is suitable to just base the app off the buttonClick sample project for how to build a table view.

**Comment by mrwhale** [comment]
Okay, if no one else has looked into this I will make it my job to finally sit down and get something going!

**Comment by eliotstocker** [comment]
thst would be great, my time working on this project has been a little limited of late and some further work on this is certaily required. that said i hope to have more time this coming week to finalise v1.7 so that new issues can be begun.

**Comment by mrwhale** [comment]
Hey guys, been a while but was able to finally pick this up and made some headway. https://github.com/mrwhale/pebble-light-controller. at the moment its a standalone app, but once I get it stable enough I will port into this app and get things working. Probably needs some help eventual @eliotstocker figuring out how to get information like IP, Port and zone names out of the Light-Controller app so i can pass to the pebble

**Comment by NoonanMedia** [comment]
no worries, just let me know, and ill hep ya write the service, shouldn't be too different from either the API or the android Wear Service

**Comment by eliotstocker** [comment]
oops wrong account, that was me! ^ haha

**Comment by eliotstocker** [comment]
@mrwhale any progress on this?

**Comment by mrwhale** [comment]
Hey @eliotstocker. No progress yet, got sidetracked a bit. But back on board now. Have just pulled the latest upstream from you and going to get to work on implementing this. Will create a branch in my fork and set up Android studio again! I've got a basic UI and watch to phone communication happening (from my last attempt) so will work on getting this talking with the light-controller app. Will definitely need some help in the design /UI area though to make things look pretty on the watch haha

**Comment by mrwhale** [comment]
Okay. So I have a working/beta version of pebble support in the app now! Let me know if you want the apk/pbw to test. Next step is to get a PR into master so we can push it it the beta users,and getting the pebble app up on the pebble store

@eliotstocker one thing I was thinking of adding was a boolean option in settings to turn pebble support on/off (so then I can check this in code and NOT run it) but will certainly need your help adding this in. For now I will just check if pebble is connected and then ignore if not

**Comment by mrwhale** [comment]
So I just changed it a little bit so instead of toggling on/off status with the select button on the pebble. I've made it so single click = on, and long press = off

@eliotstocker ready for you to check it out/merge into another branch of yours (I stupidly started working on it in my master branch :/ ) Would be awesome to get this out to Beta testers, and I get the pebble app uploaded to the pebble store as soon as we get some icons organised!

You can check the code out for the pebble app here - https://github.com/mrwhale/pebble-light-controller

but will also include in this project too shortly

**Comment by dudeofawesome** [comment]
@mrwhale, you could make a new branch off of your master for the pull request. I made the same mistake on my last PR.

**Comment by eliotstocker** [comment]
@mrwhale can you send a pull request?

**Comment by mrwhale** [comment]
Thanks @dudeofawesome! Yeah @eliotstocker I've just branched my master and created a Pull request. Hopefully that makes it a bit cleaner

**Comment by eliotstocker** [comment]
coolio :)

i have just uploaded the assets that your requested (off github) in the repo folder pebble_assets

**Comment by mrwhale** [comment]
Awesome! Much better the the on I tried to make :/ 

Sooo to post on the pebble store we can also include: Header Images, screenshots and Marketting Banner. If you are keen, can you create something for it..? (i'll create the screenshots)

![screenshot from 2016-06-24 08-49-57](https://cloud.githubusercontent.com/assets/7307429/16322748/b59c64a8-39e8-11e6-8300-5b970b0fa420.png)

https://developer.pebble.com/guides/appstore-publishing/preparing-a-submission/

**Comment by eliotstocker** [comment]
id say, lets get it up there, and im sure we can add assets later :)

**Comment by mrwhale** [comment]
Sounds like a plan. Well It's ready to go live on the pebble store whenever 1.8 goes like on the beta channel

**Comment by eliotstocker** [comment]
cool, will pull your PR tomorrow and get a beta up :)

**Comment by mrwhale** [comment]
Hey @eliotstocker! Did we get v1.8 to the beta channel?

**Comment by eliotstocker** [comment]
sorry, small issue with the upload that i didn't notice, fixed now, should be up in the next couple of hours

**Comment by mrwhale** [comment]
Wooo! Thanks:)

**Comment by mrwhale** [comment]
I'd say this issue is closed, now that its working in the beta build. There are a few fixes/features I am planning to add, but have opened these on the pebble app repo so its not clogging this one up https://github.com/mrwhale/pebble-light-controller/issues

**Comment by eliotstocker** [comment]
cool :)
